### PR TITLE
Add support for specifying Microsoft.NET.Sdk.Web as the sdk

### PR DESCRIPTION
**Pull request by seesharper**
This PR adds support for specifying an SDK using the `#r` directive.
The feature is restricted to `Microsoft.NET.Sdk.Web` at the moment.

Example 
```csharp
#r "sdk:Microsoft.NET.Sdk.Web"

using Microsoft.AspNetCore.Builder;

var a = WebApplication.Create();
a.MapGet("/", () => "Hello world");
a.Run();
```
This PR also includes a number of updates to NuGet packages referenced by `dotnet-script`. This should minimize the need for `--isolated-load-context`. Also removed a few dependencies that are no longer needed. Some of them was just referenced to make things work back when we were at netcoreapp2.1. 

Other news. Removed support for Chocolatey 😃

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,19 +8,19 @@ Run C# scripts from the .NET CLI, define NuGet packages inline and edit/debug th
...
+| ------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------- |
+| `dotnet-script` (global tool)         | [![Nuget](http://img.shields.io/nuget/v/dotnet-script.svg?maxAge=10800)](https://www.nuget.org/packages/dotnet-script/)                                             | `net6.0`, `net5.0`, `netcoreapp3.1` |
+| `Dotnet.Script` (CLI as Nuget)        | [![Nuget](http://img.shields.io/nuget/v/dotnet.script.svg?maxAge=10800)](https://www.nuget.org/packages/dotnet.script/)                                             | `net6.0`, `net5.0`, `netcoreapp3.1` |
+| `Dotnet.Script.Core`                  | [![Nuget](http://img.shields.io/nuget/v/Dotnet.Script.Core.svg?maxAge=10800)](https://www.nuget.org/packages/Dotnet.Script.Core/)                                   | `netcoreapp3.1` , `netstandard2.0`  |
```
Comment (filipw) [inline README.md:15]: I think this should be marked as `net6.0` not 3.1

File: src/Dotnet.Script.Tests/ScriptExecutionTests.cs [hunk 2]
```diff
@@ -21,8 +21,9 @@ public ScriptExecutionTests(ITestOutputHelper testOutputHelper)
         [Fact]
         public void ShouldExecuteHelloWorld()
         {
-            var (output, _) = ScriptTestRunner.Default.ExecuteFixture("HelloWorld", "--no-cache");
-            Assert.Contains("Hello World", output);
+            ScriptTestRunner.ExecuteFixtureInProcess("HelloWorld", "--no-cache");
+            // var (output, _) = ScriptTestRunner.Default.ExecuteFixture("HelloWorld", "--no-cache");
+            // Assert.Contains("Hello World", output);
```
Comment (filipw) [inline src/Dotnet.Script.Tests/ScriptExecutionTests.cs:26]: what happened here?

File: README.md [hunk 3]
```diff
@@ -569,6 +565,24 @@ We will also see this when working with scripts in VS Code under the problems pa
...
+#r "sdk:Microsoft.NET.Sdk.Web"
+
+using Microsoft.AspNetCore.Builder;
+
+var a = WebApplication.Create();
+a.MapGet("/", () => "Hello world");
+a.Run();
+```
+
+> Please note the the only SDK currently supported is `Microsoft.NET.Sdk.Web`
```
Comment (filipw) [inline README.md:584]: different project SDKs have [different default namespaces](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/overview#implicit-using-directives) which we do not currently do with this implementation. 

we should either have a code path that inserts different default usings based on SDK, or (which I think is fine) we can add as a note here too that this is also not supported for now

**Review by filipw — APPROVED** [review]
looks good! I just left a few small comments

File: src/Dotnet.Script.Tests/ScriptExecutionTests.cs (hunk 2 above)
Comment (seesharper) [reply src/Dotnet.Script.Tests/ScriptExecutionTests.cs:26]: Forgot about that one. Just to be able to test in process during development. Fixing

File: README.md (hunk 3 above)
Comment (seesharper) [reply README.md:584]: Yeah, I actually started on this one, but realized that we need to align this with OmniSharp. One way to solve this is to pass the usings to OmniSharp as part of the compilation dependencies. I think we will save that for another PR as this is getting big enough as it is. Creating a comment on this in the code

File: README.md (hunk 1 above)
Comment (seesharper) [reply README.md]: You´re right. Fixing

**Review by filipw — APPROVED** [review]

**Comment by filipw** [comment]
Thanks!

**Comment by ihqtim** [comment]
Great, much appreciated!

**Pull request merged**